Bottom: 597a1e2213856237ded9c329a36c2c7e7ef12195
Top:    f97f83c545f70dd6a011ad916a98bf48a5ffa59c
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-02-06 09:08:30 -0800

mac80211: Make STA disconnect messages warn instead of debug.

This makes them more easily seen in /var/log/messages
and the console, for instance.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/net/mac80211/debug.h b/net/mac80211/debug.h
index 1956b31..ed8ce2d 100644
--- a/net/mac80211/debug.h
+++ b/net/mac80211/debug.h
@@ -193,6 +193,9 @@ do {									\
 	_sdata_dbg(MAC80211_MLME_DEBUG,					\
 		   sdata, fmt, ##__VA_ARGS__)
 
+#define mlme_wrn(sdata, fmt, ...)					\
+	_sdata_err(sdata, fmt, ##__VA_ARGS__)
+
 #define mlme_dbg_ratelimited(sdata, fmt, ...)				\
 	_sdata_dbg(MAC80211_MLME_DEBUG && net_ratelimit(),		\
 		   sdata, fmt, ##__VA_ARGS__)
diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 8d426f6..c15ee2d 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -3912,7 +3912,7 @@ void ieee80211_sta_work(struct ieee80211_sub_if_data *sdata)
 					 max_tries);
 				ieee80211_mgd_probe_ap_send(sdata);
 			} else {
-				mlme_dbg(sdata,
+				mlme_wrn(sdata,
 					 "No ack for nullfunc frame to AP %pM, disconnecting.\n",
 					 bssid);
 				ieee80211_sta_connection_lost(sdata, bssid,
@@ -3922,7 +3922,7 @@ void ieee80211_sta_work(struct ieee80211_sub_if_data *sdata)
 		} else if (time_is_after_jiffies(ifmgd->probe_timeout))
 			run_again(sdata, ifmgd->probe_timeout);
 		else if (ieee80211_hw_check(&local->hw, REPORTS_TX_ACK_STATUS)) {
-			mlme_dbg(sdata,
+			mlme_wrn(sdata,
 				 "Failed to send nullfunc to AP %pM after %dms, disconnecting\n",
 				 bssid, probe_wait_ms);
 			ieee80211_sta_connection_lost(sdata, bssid,
@@ -3938,7 +3938,7 @@ void ieee80211_sta_work(struct ieee80211_sub_if_data *sdata)
 			 * We actually lost the connection ... or did we?
 			 * Let's make sure!
 			 */
-			mlme_dbg(sdata,
+			mlme_wrn(sdata,
 				 "No probe response from AP %pM after %dms, disconnecting.\n",
 				 bssid, probe_wait_ms);
